src/test/java/com/hotels/beeju/extensions/ThriftHiveMetaStoreJUnitExtensionTest.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (C) 2015-2019 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.beeju.extensions;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.hive.metastore.HiveMetaStoreClient;
+import org.apache.hadoop.hive.metastore.api.AlreadyExistsException;
+import org.apache.hadoop.hive.metastore.api.Database;
+import org.apache.hadoop.hive.metastore.api.InvalidObjectException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRule;
+import com.hotels.beeju.ThriftHiveMetaStoreJUnitRuleTest;
+
+public class ThriftHiveMetaStoreJUnitExtensionTest implements BeforeEachCallback {
+
+  private static File defaultTempRoot;
```

Comment:
Is there a reason for these two being static? If not we generally favour not using static (for various reasons we can go into).

Reply:
The temp folders aren't actually needed anymore for these tests either so I'll remove them, but happy to learn more about static vs non-static :D